### PR TITLE
Run3-hcx242 First step to make HCAL parameters to be read in DDD/DD4Hep

### PR DESCRIPTION
**Pull request by bsunanda**
#### PR description:

First step to load HcalParameters from the xml files through DDD as well as DD4Hep scenario

#### PR validation:

Use standard runTheMatrix.py

#### if this PR is a backport please specify the original PR:

Nothing special

### CODE REVIEW

**Comment by bsunanda** [comment]
@cmsbuild Please test

**Comment by bsunanda** [comment]
@cmsbuild Please test

File: Geometry/HcalCommonData/src/HcalGeomParameters.cc [hunk 1]
```diff
@@ -15,7 +15,7 @@
 #include "CLHEP/Units/GlobalPhysicalConstants.h"
 #include "CLHEP/Units/GlobalSystemOfUnits.h"
 
-//#define EDM_ML_DEBUG
+#define EDM_ML_DEBUG
```
Comment (civanch) [inline Geometry/HcalCommonData/src/HcalGeomParameters.cc]: @bsunanda , here debug mode left in the PR

File: Geometry/HcalCommonData/src/HcalParametersFromDD.cc [hunk 2]
```diff
@@ -160,7 +163,6 @@ bool HcalParametersFromDD::build(const DDCompactView* cpv, HcalParameters& php)
     std::cout << " [" << ++i << "] = " << (*it) / CLHEP::deg;
```
Comment (civanch) [inline Geometry/HcalCommonData/src/HcalParametersFromDD.cc]: @bsunanda , why we are using std::cout even in the debug mode?

File: Geometry/HcalCommonData/plugins/HcalParametersESModule.cc [hunk 3]
```diff
@@ -5,16 +5,18 @@
...
 #include "CondFormats/GeometryObjects/interface/HcalParameters.h"
 #include "DetectorDescription/Core/interface/DDCompactView.h"
+#include "DetectorDescription/DDCMS/interface/DDCompactView.h"
 #include "Geometry/Records/interface/IdealGeometryRecord.h"
 #include "Geometry/Records/interface/HcalParametersRcd.h"
 #include "Geometry/HcalCommonData/interface/HcalParametersFromDD.h"
 
 #include <memory>
 
+#define EDM_ML_DEBUG
```
Comment (civanch) [inline Geometry/HcalCommonData/plugins/HcalParametersESModule.cc]: @bsunanda , debug mode is left...

File: Geometry/HcalCommonData/src/HcalParametersFromDD.cc (hunk 2 above)
Comment (bsunanda) [reply Geometry/HcalCommonData/src/HcalParametersFromDD.cc]: I have changed a few. The rest will be taken care of when I complete the dd4hep part.

File: Geometry/HcalCommonData/plugins/HcalParametersESModule.cc (hunk 3 above)
Comment (bsunanda) [reply Geometry/HcalCommonData/plugins/HcalParametersESModule.cc]: taken care of

File: Geometry/HcalCommonData/src/HcalGeomParameters.cc (hunk 1 above)
Comment (bsunanda) [reply Geometry/HcalCommonData/src/HcalGeomParameters.cc]: Taken care of

**Comment by bsunanda** [comment]
@cmsbuild Please test

File: Geometry/HcalCommonData/src/HcalParametersFromDD.cc [hunk 4]
```diff
@@ -124,44 +125,79 @@ bool HcalParametersFromDD::build(const DDCompactView* cpv, HcalParameters& php)
...
+  }
+  edm::LogVerbatim("HCalGeom") << "HcalParametersFromDD:phibin: " << php.phibin.size() << ": " << ss6.str();
+  std::stringstream ss7;
+  for (unsigned int it = 0; it < php.phitable.size(); it++) {
+    if (it / 10 * 10 == it) {
+      ss7 << "\n";
+    }
+    ss7 << " [" << it << "] " << convertRadToDeg(php.phitable[it]);
+  }
+  edm::LogVerbatim("HCalGeom") << "HcalParametersFromDD:phitable: " << php.phitable.size() << ": " << ss7.str();
```
Comment (fabiocos) [inline Geometry/HcalCommonData/src/HcalParametersFromDD.cc]: @bsunanda sorry, why not getting rid of *all* the ```std::cout``` in this function?

**Comment by fabiocos** [comment]
+1

**Comment by bsunanda** [comment]
@fabiocos I had almost no internet connection those days. I went to one institute and used the limited time which I had to make some changes. The rest will come with a new PR

**Pull request merged**